src/main/scala/epic/parser/models/PositionalNeuralModel.scala

```diff
@@ -0,0 +1,501 @@
+//          for (begin <- 0 until words.size) {
+//            for (end <- begin + 1 to words.size) {
+//              if (constraints.isAllowedSpan(begin, end)) {
+//                for (split <- begin + 1 until end) {
+//                  if (constraints.isAllowedSpan(begin, split) && constraints.isAllowedSpan(split, end)) {
+//                    surfaceFeatsSeen += surffeat.featuresForSplit(begin, split, end)
+//                  }
+//                }
+//              }
+//            }
+//          }
+//          surfaceFeatsSeen
+//        })
+////        println(data.size + " examples in mini-batch yield " + fvs.size + " feature vectors to feedforward")
+//        grammar.layers(0).applyBatchNormalization(fvs)
+//      }
+    }
+  }
+
+  @SerialVersionUID(4749637878577393596L)
+  class PositionalNeuralGrammar[L, L2, W](val topology: RuleTopology[L],
+                                             val lexicon: Lexicon[L, W],
+                                             val refinedTopology: RuleTopology[L2],
+                                             val refinements: GrammarRefinements[L, L2],
+                                             labelFeaturizer: RefinedFeaturizer[L, W, Feature],
+                                             val surfaceFeaturizer: Word2VecSurfaceFeaturizerIndexed[W],
+                                             depFeaturizer: Word2VecDepFeaturizerIndexed[W],
+                                             val layers: IndexedSeq[OutputTransform[Array[Int],DenseVector[Double]]#OutputLayer],
+                                             penultimateLayers: IndexedSeq[epic.dense.Transform.Layer[Array[Int],DenseVector[Double]]],
+                                             depLayers: IndexedSeq[OutputTransform[Array[Int],DenseVector[Double]]#OutputLayer],
+                                             val maybeSparseSurfaceFeaturizer: Option[IndexedSpanFeaturizer[L, L2, W]],
+                                             decoupledLayers: IndexedSeq[OutputTransform[Array[Int],DenseVector[Double]]#OutputLayer],
+                                             penultimateDecoupledLayers: IndexedSeq[epic.dense.Transform.Layer[Array[Int],DenseVector[Double]]],
+                                             val weights: DenseVector[Double],
+                                             val origPTModel: PositionalNeuralModel[L,L2,W]) extends Grammar[L, W] with Serializable {
+
+    val SpanLayerIdx = 0
+    val UnaryLayerIdx = 1
+    val BinaryLayerIdx = 2
+    val dcSpanFeatOffset = layers.map(_.index.size).foldLeft(0)(_ + _) + depLayers.map(_.index.size).foldLeft(0)(_ + _)
+    val dcUnaryFeatOffset = dcSpanFeatOffset + (if (decoupledLayers.size > 0) decoupledLayers(0).index.size else 0)
+    val dcBinaryFeatOffset = dcUnaryFeatOffset + (if (decoupledLayers.size > 0) decoupledLayers(1).index.size else 0)
+    
+    override def withPermissiveLexicon: Grammar[L, W] = {
```

Comment:
i forgot what I added withPermissiveLexicon for. I should go back and figure it out and maybe delete it.